src/legacy/ui/public/vis/editors/default/agg_param.js

```diff
@@ -100,9 +109,14 @@ uiModules
             }
           };
 
+          $scope.setTouched = () => {
+            ngModelCtrl.$setTouched();
+            $scope.isInvalid = !$scope.paramValue;
+          };
+
           $scope.setValidity = (isValid) => {
             if(ngModelCtrl) {
-              $scope.isInvalid = !isValid;
+              $scope.isInvalid = ngModelCtrl.$touched ? !isValid : false;
```

Comment:
This is not working correctly. Now fields that are not using `setTouched()` are never getting the correct invalid state anymore. An easy solution would be to at least call `ngModelCtrl.$setTouched()` in line 108 when changing the value for all fields, since we can be sure the field is now touched too.